src/core/thread/csl_tx_scheduler.cpp

```diff
@@ -133,7 +132,7 @@ void CslTxScheduler::RescheduleCslTx(void)
             continue;
         }
 
-        delay = GetNextCslTransmissionDelay(child, radioNow, cslTxDelay);
```

Comment:
Just to mention that I used same `radioNow` for each child so that we can compare each child's next csl tx delay based on same `Now` value and save the cost for calling radio API. Not sure how much time would pass after processing one child (I believe it's short enough to ignore). Anyway, I agree with the new approach.